rewriter/DefaultArgs.cc

```diff
@@ -181,24 +181,35 @@ class DefaultArgsWalk {
                             if (sig == nullptr) {
                                 continue;
                             }
-                            newMethods.emplace_back(move(sig));
+                            newMethodsAndSigs.emplace_back(move(sig));
                         }
                         auto defaultArgDef = ast::MK::SyntheticMethod(loc, core::Loc(ctx.file, loc), name,
                                                                       std::move(args), std::move(rhs));
                         {
                             auto &defaultDef = ast::cast_tree_nonnull<ast::MethodDef>(defaultArgDef);
                             defaultDef.flags.isSelfMethod = mdef->flags.isSelfMethod;
                         }
-                        newMethods.emplace_back(move(defaultArgDef));
+                        newMethodsAndSigs.emplace_back(move(defaultArgDef));
                     }
                     lastSig = nullptr;
                 },
 
                 [&](ast::Expression *expr) {});
         }
 
-        for (auto &stat : newMethods) {
-            klass.rhs.emplace_back(move(stat));
+        for (auto &stat : newMethodsAndSigs) {
+            if (auto mdef = ast::cast_tree<ast::MethodDef>(stat)) {
+                auto loc = mdef->loc;
+                auto name = mdef->name;
+                auto keepName = mdef->flags.isSelfMethod ? core::Names::keepSelfDef() : core::Names::keepDef();
+
+                klass.rhs.emplace_back(move(stat));
+                klass.rhs.emplace_back(ast::MK::Send2(loc,
+                                                      ast::MK::Constant(loc, core::Symbols::Sorbet_Private_Static()),
+                                                      keepName, ast::MK::Self(loc), ast::MK::Symbol(loc, name)));
```

Comment:
Could you add a comment about why the `keepDef` calls are generated here? I think that this is the right trade-off to make, but it would be good to document it.